alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/wormhole/implementation/adapter/ComponentViewType.java

```diff
@@ -67,7 +68,7 @@ public boolean equals(final Object o) {
             return false;
         }
         final ComponentViewType nvt = (ComponentViewType) o;
-        return getWidth() == nvt.getWidth() && getHeight() == nvt.getHeight();
+        return Math.abs(getWidth() - nvt.getWidth()) < 0.0000001 && Math.abs(getHeight() - nvt.getHeight()) < 0.0000001;
```

Comment:
what is 0.00000001 or whatever? How did you compute it? What's the purpose of re-defining equals in this class anyway?

Reply:
From http://findbugs.sourceforge.net/bugDescriptions.html#FE_FLOATING_POINT_EQUALITY:

> This operation compares two floating point values for equality. Because floating point calculations may involve rounding, calculated float and double values may not be accurate. For values that must be precise, such as monetary values, consider using a fixed-precision type such as BigDecimal. For values that need not be precise, consider comparing for equality within some range, for example: if ( Math.abs(x - y) < .0000001 ). See the Java Language Specification, section 4.2.4.

Thinking about it now, it would be more appropriate to use Double.MIN_VALUE, right?

Reply:
Yes, see if it works with that